From 313d0e15c1c626bc868ef8040b25a57867f8b6cb Mon Sep 17 00:00:00 2001
From: Simon Cook <simon.cook@embecosm.com>
Date: Fri, 12 Dec 2014 15:52:23 +0000
Subject: [PATCH 19/33] Link with cachemanager.o for Epiphany when -fpic is
 used.

	* config/epiphany/epiphany.h (ENDFILE_SPEC): Add cachemanager.o
	when -fpic is used.
---
 gcc/ChangeLog.epiphany         | 5 +++++
 gcc/config/epiphany/epiphany.h | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index ad3144cbc1b..89a6eb29e90 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,8 @@
+2014-12-12  Simon Cook  <simon.cook@embecosm.com>
+
+	* config/epiphany/epiphany.h (ENDFILE_SPEC): Add cachemanager.o
+	when -fpic is used.
+
 2014-12-12  Joern Rennecke  <joern.rennecke@embecosm.com>
 
 	* config/epiphany/epiphany.md (casesi_load): Use next_nonnote_insn to
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 8f6b27784a1..de877faa80b 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -55,7 +55,8 @@ along with GCC; see the file COPYING3.  If not see
   "%{m1reg-r43:crtm1reg-r43.o%s} %{m1reg-r63:crtm1reg-r63.o%s} " \
   "crtbegin.o%s"
 
-#define ENDFILE_SPEC "crtend.o%s crtn.o%s %{fsoftware-cache:cachemanager.o%s}"
+#define ENDFILE_SPEC "crtend.o%s crtn.o%s " \
+  "%{fsoftware-cache|fpic:cachemanager.o%s}"
 
 #define CC1_SPEC "%{fsoftware-cache:-fpic}"
 
-- 
2.31.1

